Add regexp function (#4145)
brimdata/zed: docs/language/functions/regexp.md

### Function

&emsp; **regexp** &mdash; perform a regular expression search on a string

### Synopsis

```
regexp(re: string, s: string) -> any
```
### Description
The _regexp_ function returns an array of strings holding the text

> of the left most match of the regular expression string re and the

Reviewer: ```suggestion
of the left most match of the regular expression string `re` and the
```

> results of its sub expression if there are any. A null value indicates

Reviewer: ```suggestion
matches of each parenthesized subexpression (also known as capturing groups)
if there are any. A null value indicates
```

no match.

### Examples

Regexp returns an array of the match and its subexpressions:
```mdtest-command

> echo '"seafood fool"' | zq -z 'yield regexp("foo(.?)", this) -'

Reviewer: ```suggestion
echo '"seafood fool friend"' | zq -z 'yield regexp("foo(.?) (\\w+) fr.*", this)' -
```

I think it'd be helpful for the example to show multiple capture groups to help users recognize that the approach is very similar to what they may have seen in Python, TCL, etc.

```
=>
```mdtest-output

> ["food", "d"]

Reviewer: ```suggestion
["food fool friend","d","fool"]
```
